test/distributed/_tensor/test_attention.py

```diff
@@ -40,137 +38,146 @@
 
 
 c10d_functional = torch.ops.c10d_functional
+backends = []
+if PLATFORM_SUPPORTS_FLASH_ATTENTION:
+    backends.append(SDPBackend.FLASH_ATTENTION)
+if PLATFORM_SUPPORTS_MEM_EFF_ATTENTION:
+    backends.append(SDPBackend.EFFICIENT_ATTENTION)
 
 
 class RingAttentionTest(DTensorTestBase):
     @property
     def world_size(self) -> int:
-        return 2
+        return torch.cuda.device_count()
 
     @skip_if_lt_x_gpu(2)
     @skipIfRocm  # Missing _c10d_functional_autograd::all_to_all_single
     @unittest.skipIf(
-        not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Does not support flash attention"
+        not PLATFORM_SUPPORTS_FUSED_ATTENTION,
+        "Does not support flash nor efficient attention",
     )
     @with_comms
     @parametrize("is_causal", [True, False])
-    def test_ring_attention_sdpa(self, is_causal: bool) -> None:
-        device_mesh = DeviceMesh(
-            self.device_type,
-            torch.arange(0, self.world_size),
-        )
+    @parametrize("compiled", [True, False])
+    @parametrize("backend", backends)
+    def test_ring_attention_sdpa(
+        self, is_causal: bool, compiled: bool, backend: SDPBackend
+    ) -> None:
+        device_mesh = DeviceMesh(self.device_type, torch.arange(0, self.world_size))
         dtype = torch.bfloat16
         bs = 8
-        query_tokens = 8
-        context_tokens = query_tokens if is_causal else 8
+        query_tokens = 64
+        context_tokens = 64
         dim = 32
         nheads = 8
-        query = torch.rand(
+        torch.manual_seed(10)
+        dtype = (
+            torch.bfloat16 if backend == SDPBackend.FLASH_ATTENTION else torch.float32
+        )
+
+        if is_causal and compiled and self.world_size > 2:
+            # TODO: Fix this after we move `wait_tensor` to use `with_effect`.
+            return
```

Comment:
do we know what kind of error it throwing? if so we could use `self.assertRaises` instead

Reply:
The error can be random, either accuracy or timeout. So we could not use `assertRaises` at this moment.